server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregator.java

```diff
@@ -260,82 +268,68 @@ public InternalAggregation buildEmptyAggregation() {
          */
         private long maxCost = -1;
         private long estimateCostTime;
-        private Weight[] weights;
-        /**
-         * If {@link #estimateCost} was called then this'll contain a
-         * scorer per leaf per filter. If it wasn't then this'll be {@code null}.
-         */
-        private BulkScorer[][] scorers;
         private int segmentsWithDeletedDocs;
         /**
          * Count of segments with documents have consult the {@code doc_count}
          * field.
          */
-        private int segmentsWithDocCount;
+        private int segmentsWithDocCountField;
 
         private FilterByFilter(
             String name,
-            String[] keys,
-            Query[] filters,
+            List<QueryToFilterAdapter> filters,
             boolean keyed,
             AggregationContext context,
             Aggregator parent,
             CardinalityUpperBound cardinality,
             Map<String, Object> metadata
         ) throws IOException {
-            super(name, AggregatorFactories.EMPTY, keys, keyed, null, context, parent, cardinality, metadata);
-            this.filters = filters;
+            super(name, AggregatorFactories.EMPTY, filters, keyed, null, context, parent, cardinality, metadata);
             this.profiling = context.profiling();
         }
 
         /**
          * Estimate the number of documents that this aggregation must visit. We'll
          * stop counting once we've passed {@code maxEstimatedCost} if we aren't profiling.
          */
+        @SuppressWarnings("resource") // We're not in change of anything Closeable
```

Comment:
While I was porting this to use the new `QueryToFilterAdatapter` I noticed a problem: previously this would abort estimating the cost if the cost went over the max which is probably good. And when you profile it *wouldn't* abort the estimating so that you could get back what the "real total" estimate would be. Also probably good. But we timed how long the estimate took. And previously the timer included how long it took to estimate without the early abort. That felt wrong. Now we stop the timer as soon as we pass the limit. Or when we return.